components/selection/selection-mixin.js

```diff
@@ -69,24 +80,33 @@ export const SelectionMixin = superclass => class extends RtlMixin(superclass) {
 
 	disconnectedCallback() {
 		super.disconnectedCallback();
+		if (this._observerController) this._observerController.hostDisconnected();
+		if (this._selectablesController) this._selectablesController.hostDisconnected();
+
 		if (this.selectionSingle) this.removeEventListener('keydown', this._handleRadioKeyDown);
 		if (this.selectionSingle) this.removeEventListener('keyup', this._handleRadioKeyUp);
 		this.removeEventListener('d2l-selection-change', this._handleSelectionChange);
-		this.removeEventListener('d2l-selection-observer-subscribe', this._handleSelectionObserverSubscribe);
-		this.removeEventListener('d2l-selection-input-subscribe', this._handleSelectionInputSubscribe);
+	}
+
+	getController(controllerId) {
```

Comment:
I don't love this, but I couldn't think of another great way to do both:
1. Expose the controller without a forced naming convention AND
2. Have the ability for there to be multiple provider controllers in a single component